### DISCUSSION TITLE
Webhook not hitting the job

### DISCUSSION THREAD

**Discussion by JGarciaGuerrero** [spatie/laravel-stripe-webhooks#146] [Q&A]
Hi, there, I´m using this app to handle my wehbhooks and I´ve found an issue, the config webhooks doesn´t hit the jobs (at least in localhost).

I´ve followed the docs, installed and published and make the route:

<img width="245" alt="image" src="https://user-images.githubusercontent.com/36189931/209807832-e9a1c130-0529-4dbc-8276-91109d75c3ce.png">

Make the exception in csrf

<img width="161" alt="image" src="https://user-images.githubusercontent.com/36189931/209807875-4c0ae791-197e-45e7-868b-396fda191723.png">

In my config I have this: 
<img width="551" alt="image" src="https://user-images.githubusercontent.com/36189931/209810003-3d6d67ef-55d1-423e-bd59-cdc575eda65b.png">

And I created a job using the artisan command, but when in my local cmd I try the events like: stripe trigger charge.succeded

it gives me always a 200 response, no matter if I even delete all the content in the config/stripe-webhooks: 
<img width="306" alt="image" src="https://user-images.githubusercontent.com/36189931/209808565-5d172920-8bb4-48ad-8add-cc2aa412e6f2.png">

Even with this, I´m getting a 200 OK response.

But if I delete the "return" as well, it gives me a 500 response, so I guess the route is going well to config, but for some reason it didn´t matter what I return meanwhile I return an array.

I´ve looking in this github and some people had the same problem but no one has given a response yet.

Thanks for your time!

**Comment by JGarciaGuerrero** [comment]
> Hi, there, I´m using this app to handle my wehbhooks and I´ve found an issue, the config webhooks doesn´t hit the jobs (at least in localhost).
> 
> I´ve followed the docs, installed and published and make the route:
> 
> <img alt="image" width="245" src="https://user-images.githubusercontent.com/36189931/209807832-e9a1c130-0529-4dbc-8276-91109d75c3ce.png">
> 
> Make the exception in csrf
> 
> <img alt="image" width="161" src="https://user-images.githubusercontent.com/36189931/209807875-4c0ae791-197e-45e7-868b-396fda191723.png">
> 
> In my config I have this: <img alt="image" width="545" src="https://user-images.githubusercontent.com/36189931/209808134-31cc07ad-8a04-43f8-8320-5fbe34a752e7.png">
> 
> And I created a job using the artisan command, but when in my local cmd I try the events like: stripe trigger charge.succeded
> 
> it gives me always a 200 response, no matter if I even delete all the content in the config/stripe-webhooks: <img alt="image" width="306" src="https://user-images.githubusercontent.com/36189931/209808565-5d172920-8bb4-48ad-8add-cc2aa412e6f2.png">
> 
> Even with this, I´m getting a 200 OK response.
> 
> But if I delete the "return" as well, it gives me a 500 response, so I guess the route is going well to config, but for some reason it didn´t matter what I return meanwhile I return an array.
> 
> I´ve looking in this github and some people had the same problem but no one has given a response yet.
> 
> Thanks for your time!

Pd: The tables in my ddbb are filling correctly, jobs and webhooks events, but the action in the job is not working, even trying to return an error from the job, it always shows 200 OK

**Reply by cyrillkalita** [reply]
For anyone who finds this question and is befuddled:

Assuming you:
- set up local webhooks properly: _have the db ready, have registered the webhook route, have excluded it from CSRF protection_
- properly configured jobs: _for `invoice.payment_succeeded` would be a `invoice_payment_succeeded` job or `stripe-webhooks::invoice.payment_succeeded` event_
- are listening with local stripe CLI for events: _you see the events coming in in your terminal_
- and you see the webhook calls persisted in the DB, but not processed: 

**Remember**: if your jobs `ShouldQueue`, they are. They be placed in the default stripe queue - which you _think_ might be `sync` but check the value of `STRIPE_WEBHOOK_CONNECTION` - and if you see `redis`, just swap it for `sync`.

Go conquer the world!
